pilot/pkg/model/service.go

```diff
@@ -366,24 +366,12 @@ func WorkloadInstancesEqual(first, second *WorkloadInstance) bool {
 	if first.Kind != second.Kind {
 		return false
 	}
-	if !portMapEquals(first.PortMap, second.PortMap) {
+	if !maps.Equal(first.PortMap, second.PortMap) {
 		return false
 	}
 	return true
 }
 
-func portMapEquals(a, b map[string]uint32) bool {
```

Comment:
`maps.Equal` checks whether the key exists, but `portMapEquals` does not. This causes the following examples to have different results.

```code
a := map[string]uint32{"a": 0}
b := map[string]uint32{"b": 0}

portMapEquals(a, b) // got: true
maps.Equal(a, b)    // got: false
````

Reply:
I don't think there is a case where the value of the PortMap, which can also be described as the port, should be 0.

Reply:
I don't think there is a case where the value of the PortMap, which can also be described as the port, should be 0.

Reply:
You can think that `portMapEquals` is not right